### PR TITLE
Add Fractional Second Digits to neo datetime

### PR DESCRIPTION
**Pull request by sffc**
#1317

Note: For now I'm adding them as an option on the Second field but I think I'll probably make a separate field as we're discussing in the spec draft.

### CODE REVIEW

**Comment by sffc** [comment]
Seems to be passing CI. Ready to review for whoever gets to it first.

File: components/datetime/src/format/mod.rs [hunk 1]
```diff
@@ -2,8 +2,20 @@
...
 pub mod datetime;
 #[cfg(feature = "experimental")]
 pub mod neo;
 pub mod time_zone;
 pub mod zoned_datetime;
+
+/// Internal non-pattern runtime options passed to the formatter
+#[derive(Debug, Copy, Clone, Default)]
+pub(crate) struct FormattingOptions {
+    #[cfg(feature = "experimental")]
```
Comment (Manishearth) [inline components/datetime/src/format/mod.rs:17]: question: this goes away with neo, yes?

File: components/datetime/src/format/datetime.rs [hunk 2]
```diff
@@ -668,69 +675,66 @@ where
...
                         let mut s = FixedDecimal::from(usize::from(second));
                         let _infallible = s.concatenate_end(
                             FixedDecimal::from(usize::from(ns)).multiplied_pow10(-9),
                         );
                         debug_assert!(_infallible.is_ok());
-                        s.pad_end(-(p as i16));
+                        s.pad_end(position);
+                        if maybe_next_field.is_none() {
+                            // Truncate on semantic option but not "S" field
+                            // TODO: Does this make sense?
```
Comment (Manishearth) [inline components/datetime/src/format/datetime.rs:724]: (I don't understand the context here)

File: components/datetime/src/format/mod.rs (hunk 1 above)
Comment (Manishearth) [reply components/datetime/src/format/mod.rs:17]: Since this is a new type I'd kinda prefer to avoid experimental-gating this temporarily

**Review by Manishearth — APPROVED** [review]

File: components/datetime/src/format/datetime.rs (hunk 2 above)
Comment (sffc) [reply components/datetime/src/format/datetime.rs:724]: There are two ways to hit this code path

1. A pattern such as "H:mm:ssSSS"
2. fractionalSecondDigits option

The behavior implemented for "SSS" had been that it is a minimum but not maximum number of digits. 

I'm currently implementing fSD as both minimum and maximum.

**Comment by sffc** [comment]
I'm merging this to unblock me but both comments will be resolved before 2.0: the experimental feature one because we'll be pulling it out of experimental soon, and the TODO one because I'll do a scrub of TODOs when graduating the code.

**Pull request merged**